### ISSUE TITLE
resolvers: add support for IBM Spectrum LSF

### ISSUE THREAD

**Issue by wihobbs** [flux-framework/flux-core#4377]
**tl;dr This issue is opened to track some things I'm trying that will expand flux proxy/flux uri support to allow users to work with Flux interactively on LSF-based systems.**

I've put some time into this and after consulting with @jameshcorbett we decided that opening an issue would be a good way to track my progress and summarize some coffee call/Slack conversations with @grondo.

Currently, we have a resolver for Slurm that leverages `srun`'s job attachment feature (`srun --overlap --jobid=`) to obtain the pid's of individual processes running within a job and [resolve the URIs of those proceses](https://github.com/flux-framework/flux-core/blob/master/src/bindings/python/flux/uri/resolvers/slurm.py#L48). For Slurm-based machines, this works well, and allows users to connect to an existing flux session via `flux proxy`.

For LSF-based machines, it's a bit more complicated to work with Flux interactively. LSF does not offer a `--pty` equivalent for interactive sessions, so starting flux interactively is restricted to 1 node. 
```
[hobbs17@lassen709:~]$ bsub -nnodes 3 -Is -W 60 -core_isolation 2 /usr/bin/flux start
Job <3662359> is submitted to default queue <pbatch>.
<<Waiting for dispatch ...>>
<<Starting on lassen710>>
(base) [hobbs17@lassen210:~]$ flux resource list
     STATE NNODES   NCORES    NGPUS NODELIST
      free      1       40        0 lassen210
```
Or, as @ryanday36 pointed out, you can use scripts/commands to issue `jsrun` commands in an allocation session like so:
```
[day36@lassen16:~]$ jsrun -a 1 -c ALL_CPUS -g ALL_GPUS -n 1 --bind=none --smpiargs="-disable_gpu_hooks" flux start flux resource list
```

@grondo suggested that an LSF URI resolver could help with this. If a user was able to resolve URIs on LSF, they could batch a command like `flux start sleep inf` into the queue, and once their job went through, connect to the flux session from a login node using `flux proxy`.

Without the helpful `--overlap --jobid=` flags that Slurm provides, getting pids (and by extension URIs) for LSF jobs is a bit more complicated. [battach](https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=reference-battach)  _seems_ like it would support this, but the breakdown between launch and batch nodes confuses things here. 
In my flux session, on lassen649, I'm running flux:
```shell
+ exec bsub -nnodes 1 -Is -W 60 -core_isolation 2 /usr/tce/packages/lalloc/lalloc-2.0/bin/lexec flux start
Job <3662572> is submitted to default queue <pbatch>.
<<Waiting for dispatch ...>>
<<Starting on lassen710>>
<<Waiting for JSM to become ready ...>>
<<Redirecting to compute node lassen649, setting up as private launch node>>
```
But when I attach to the job, it puts me on lassen710, which is not running flux (and won't even let me do `jsrun`):
```shell
[hobbs17@lassen709:~]$ battach 3662572
Attaching job execution host: lassen710 for job: <3662572>
(base) [hobbs17@lassen710:~]$ ps
   PID TTY          TIME CMD
 12872 pts/18   00:00:00 sh
 16767 pts/18   00:00:00 ps
(base) [hobbs17@lassen710:~]$ jsrun -a 1 -c ALL_CPUS -g ALL_GPUS -n 1 --bind=none --smpiargs="-disable_gpu_hooks" flux start flux resource list
Error: It is only possible to use js commands within a job allocation unless CSM is running
06-20-2022 13:09:18:171 7420 main: Error initializing RM connection. Exiting.
(base) [hobbs17@lassen710:~]$ flux resource list
flux-resource: ERROR: [Errno 2] Unable to connect to Flux
```
Another thought I had was to use shell asterisk expansion after batch node names like so:
```shell
[hobbs17@lassen709:~]$ flux proxy ssh://lassen196/var/tmp/flux-*/local-0
(base) [hobbs17@lassen709:~]$ 
```
That worked, but Mark pointed out this won't work if there are multiple flux instances running on one host node, and he's right (regardless of whether the instances are nested or not):
```shell
[hobbs17@lassen709:~]$ flux proxy ssh://lassen196/var/tmp/flux-*/local-0
Usage: flux relay [OPTIONS] path
Relay messages between stdio and local://path
  -h, --help             Display this message.
flux-proxy: ssh://lassen196/var/tmp/flux-*/local-0: Connection reset by peer
```

Moving forward, I'm going to get some clarification from our LC support on `battach`. If that doesn't work, [individual job-level information in LSF](https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=jobs-view-information-about) implies that [specific process IDs can be accessed](https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=jobs-job-level-information), so maybe there's a workaround by trying to access job-level system information.
